pkl-core/src/main/java/org/pkl/core/externalProcess/ExternalReaderRuntime.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.externalProcess;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Objects;
+import org.msgpack.core.MessagePack;
+import org.pkl.core.externalProcess.ExternalProcessMessages.*;
+import org.pkl.core.messaging.Message.Type;
+import org.pkl.core.messaging.MessageTransport;
+import org.pkl.core.messaging.MessageTransports;
+import org.pkl.core.messaging.Messages.*;
+import org.pkl.core.messaging.ProtocolException;
+import org.pkl.core.util.Nullable;
+
+/** An implementation of the client side of the external reader flow */
+public class ExternalReaderRuntime {
```

Comment:
Let's move this and all related "runtime" classes to pkl-config-java. These are not core concerns of Pkl, and we don't need to ship these classes with pkl-cli, for example.

Reply:
This class is used in the pkl-core tests (pkl-core/src/test/kotlin/org/pkl/core/externalProcess/TestExternalProcess.kt). I'm fine moving it if the pkl-core tests can depend on pkl-config-java.